sqlglot/dialects/dialect.py

```diff
@@ -137,6 +137,13 @@ def get_start_end(token_type: TokenType) -> t.Tuple[t.Optional[str], t.Optional[
         if enum not in ("", "bigquery"):
             klass.generator_class.SELECT_KINDS = ()
 
+        if enum not in ("", "hive", "spark", "spark2"):
+            modifier_transforms = klass.generator_class.AFTER_HAVING_MODIFIER_TRANSFORMS.copy()
+            for modifier in ("cluster", "distribute", "sort"):
+                modifier_transforms.pop(modifier, None)
```

Comment:
why do you need the None here? you shouldn't be popping twice right

Reply:
The way this is currently implemented, if you load `Hive` first then it will pop these off and then `Spark2` which inherits from it won't have anything to pop, so it raises a `KeyError`.

Reply:
so then don't put spark, and just put in hive?

Reply:
Ah apologies, I got confused - we actually _don't_ wanna pop the modifiers for these dialects. The issue arises for the other dialects, e.g. you pop in postgres and so redshift will raise the key error.